sched/scheduler/recover_jobs.go

```diff
@@ -0,0 +1,131 @@
+package scheduler
+
+import (
+	"log"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/scootdev/scoot/saga"
+	"github.com/scootdev/scoot/sched"
+)
+
+// recovers all active sagas from the specified SagaCoordinator with ForwardRecovery.
+// ActiveSagas are recovered in parallel and are added to the addJobCh to be rescheduled
+// This method returns once all activeSagas have been successfully recovered.
+func recoverJobs(sc saga.SagaCoordinator, addJobCh chan jobAddedMsg) {
+
+	recoveryActiveSagaAttempts := 0
+	activeSagas, err := sc.Startup()
+	for err != nil {
+		// Retry until we succeed,  This has to eventually succeed
+		// for us to make progress.
+		// TODO: Add metrics for failure rate, this would be something we should alert on.
+		recoveryActiveSagaAttempts++
+		log.Printf("ERROR: occurred getting ActiveSagas from SagaLog %v", err)
+
+		delay := calculateExponentialBackoff(recoveryActiveSagaAttempts, time.Duration(1)*time.Minute)
+		time.Sleep(delay)
+		activeSagas, err = sc.Startup()
+	}
+
+	// if there are no activeSagas return
+	if activeSagas == nil {
+		return
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(len(activeSagas))
+
+	// Spin off separate go routines for each active saga.  This way we can
+	// do recovery in parallel by making multiple requests
+	// TODO: limit max parallel requests to RecoverSagaState?
+	for _, sagaId := range activeSagas {
+
+		go func(sagaId string) {
+			defer wg.Done()
+			activeSaga := recoverSaga(sc, sagaId)
+			if activeSaga != nil {
+				job, err := sched.DeserializeJob(activeSaga.GetState().Job())
+				if err != nil {
+					// TODO: Increment counter? A breaking change was made
```

Comment:
I dont want it to panic because it'll crash the server, and negatively impact everyone if only one job gets messed up.

It also won't get removed from the log, so it'll get retried at next server start up.  Ideally this should fire some kind of critical alert, because it'll require manual intervention.